RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelRawToClusterGPUKernel.cu

```diff
@@ -353,20 +351,25 @@ namespace pixelgpudetails {
       }
 
       uint32_t link = sipixelconstants::getLink(ww);  // Extract link
-      uint32_t roc = sipixelconstants::getROC(ww);    // Extract Roc in link
+      uint32_t roc = sipixelconstants::getROC(ww);    // Extract ROC in link
       pixelgpudetails::DetIdGPU detId = getRawId(cablingMap, fedId, link, roc);
+      uint32_t rawId = detId.rawId;
 
-      uint8_t errorType = checkROC(ww, fedId, link, cablingMap, debug);
+      uint8_t errorType = checkROC<debug>(ww, fedId, link, cablingMap);
       skipROC = (roc < pixelgpudetails::maxROCIndex) ? false : (errorType != 0);
       if (includeErrors and skipROC) {
-        uint32_t rID = getErrRawID(fedId, ww, errorType, cablingMap, debug);
+        uint32_t rID = getErrRawID<debug>(fedId, ww, errorType, cablingMap);
         err->push_back(SiPixelErrorCompact{rID, ww, errorType, fedId});
         continue;
       }
 
-      uint32_t rawId = detId.rawId;
-      uint32_t rocIdInDetUnit = detId.rocInDet;
-      bool barrel = isBarrel(rawId);
+      // check for spurious channels
+      if (roc > MAX_ROC or link > MAX_LINK) {
+        if constexpr (debug) {
+          printf("spurious roc %d found on link %d, detector %d (index %d)\n", roc, link, rawId, gIndex);
```

Comment:
rawId is "of course" wrong at this point...
one could use
```
getRawId(cablingMap, fedId, link, 1).rawId;
```
under the assumption that all roc in a link belong to the same det

Reply:
Thanks.